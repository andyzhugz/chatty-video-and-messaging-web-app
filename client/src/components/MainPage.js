import React, { useState, useEffect } from 'react'
import NaviBar from './NaviBar';
import {ChatView} from './ChatView';

function MainPage() {
  useEffect(() => {
    const loggedInUser = localStorage.getItem("curr_user");
    if (!loggedInUser || loggedInUser === "") {
      window.open("/login","_self");
    }
  }, []);
  return (
    <div id="m-parent">
      <div id="m-navi">
        <NaviBar />
      </div>
      <div id="m-content">
        <ChatView />
      </div>
    </div>
  );
}

export default MainPage;
