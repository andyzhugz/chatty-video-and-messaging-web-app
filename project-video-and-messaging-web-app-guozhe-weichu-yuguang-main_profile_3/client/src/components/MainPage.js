import React from 'react';
import NaviBar from './NaviBar';
import ChatView from './ChatView';

function MainPage() {
  return (
    <div id="m-parent">
      <div id="m-navi">
        <NaviBar />
      </div>
      <div id="m-content">
        <ChatView />
      </div>
    </div>
  );
}

export default MainPage;
