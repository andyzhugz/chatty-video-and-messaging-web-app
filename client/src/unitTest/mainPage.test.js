import React from 'react';
import MainPage from '../components/mainPage.js';

const renderer = require('react-test-renderer');

// snapshot testings
describe('Test snapshot', () => {
  test('Test Profile', () => {
    const component = renderer.create(<MainPage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});