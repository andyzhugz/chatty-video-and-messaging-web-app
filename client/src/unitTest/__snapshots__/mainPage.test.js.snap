// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Test Profile 1`] = `
<div
  id="m-parent"
>
  <div
    id="m-navi"
  >
    <div>
      <ul
        id="nav_ul"
      >
        <li
          className="nav_li"
        >
          <a
            href="/profile"
          >
            Profile
          </a>
        </li>
        <li
          className="nav_li"
        >
          <a
            href="/main"
          >
            Messaging
          </a>
        </li>
        <li
          className="nav_li"
        >
          <a
            href="/status"
          >
            Status
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    id="m-content"
  >
    <div
      id="container"
    >
      <aside>
        <header>
          <input
            id="myInput"
            onKeyUp={[Function]}
            placeholder="search"
            type="text"
          />
        </header>
        <ul
          id="myul"
        >
          <li
            onClick={[Function]}
            value="Any"
          >
            <h2>
              Any
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Bob"
          >
            <h2>
              Bob
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Cat"
          >
            <h2>
              Cat
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Fish"
          >
            <h2>
              Fish
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Tiger"
          >
            <h2>
              Tiger
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Pig"
          >
            <h2>
              Pig
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Deer"
          >
            <h2>
              Deer
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Snake"
          >
            <h2>
              Snake
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Iiger"
          >
            <h2>
              Iiger
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Lion"
          >
            <h2>
              Lion
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Bear"
          >
            <h2>
              Bear
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Bird"
          >
            <h2>
              Bird
            </h2>
          </li>
          <li
            onClick={[Function]}
            value="Monkey"
          >
            <h2>
              Monkey
            </h2>
          </li>
        </ul>
      </aside>
      <main>
        <header>
          <div>
            <h2
              id="chat_title"
            >
              Vincent Porter
            </h2>
          </div>
        </header>
        <ul
          id="chat"
        >
          <li
            class="you"
          >
            <div
              class="entete"
            >
              <span
                class="status green"
              />
              <h2>
                Vincent
              </h2>
              <h3>
                10:12AM, Today
              </h3>
            </div>
            <div
              class="message"
            >
              Hi, this is the first message
            </div>
          </li>
          <li
            class="you"
          >
            <div
              class="entete"
            >
              <span
                class="status green"
              />
              <h2>
                Vincent
              </h2>
              <h3>
                10:12AM, Today
              </h3>
            </div>
            <div
              class="message"
            >
              Hello
            </div>
          </li>
        </ul>
        <div
          id="btu_div"
        >
          <button
            className="func_btu"
          >
            Video Call
          </button>
          <button
            className="func_btu"
          >
            <label
              class="custom-file-upload"
              for="image-upload"
            >
              <i
                class="fa fa-cloud-upload"
              />
               Send Image
            </label>
            <input
              accept="image/png, image/jpg"
              id="image-upload"
              onChange={[Function]}
              type="file"
            />
          </button>
          <button
            className="func_btu"
          >
            <label
              class="custom-file-upload"
              for="audio-upload"
            >
              <i
                class="fa fa-cloud-upload"
              />
               Send Audio
            </label>
            <input
              accept="audio/mp3"
              id="audio-upload"
              onChange={[Function]}
              type="file"
            />
          </button>
          <button
            className="func_btu"
            onClick={[Function]}
          >
            Record Audio
          </button>
          <button
            className="func_btu"
          >
            <label
              class="custom-file-upload"
              for="video-upload"
            >
              <i
                class="fa fa-cloud-upload"
              />
               Send Video
            </label>
            <input
              accept="video/mp4"
              id="video-upload"
              onChange={[Function]}
              type="file"
            />
          </button>
        </div>
        <footer>
          <textarea
            id="textarea"
            placeholder="Type your message"
          />
          <a
            href="#"
            id="submitbtu"
            onClick={[Function]}
          >
            Send
          </a>
        </footer>
        <div
          class="overlay"
          id="popup1"
        >
          <div
            class="popup"
          >
            <h2>
              Record
            </h2>
            <a
              class="close"
              onClick={[Function]}
            >
              Ã—
            </a>
            <p>
              <button
                id="startRecord"
                onClick={[Function]}
              >
                start
              </button>
              <button
                id="stopRecord"
                onClick={[Function]}
              >
                stop
              </button>
              <button
                id="sendRecord"
                onClick={[Function]}
              >
                SEND
              </button>
            </p>
            <p>
              <audio
                id="recordedAudio"
              />
            </p>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
